src/providers/twitch/IrcMessageHandler.cpp

```diff
@@ -21,6 +21,101 @@
 
 namespace chatterino {
 
+static float LongestCommonSubstring(QString str1, QString str2)
```

Comment:
This should probably be put into the unnamed namespace and the parameters can be `const &`.

Also, I find the naming a tad bit confusing since this not actually returning the longest common substring. Maybe something like `relativeSimilarity` would be better?

Reply:
yeah i wasnt sure where to put this, and also where to put the actual `IrcMessageHandler::similarity` functions
the parameters should be const & (tho iirc qstrings are COW, but yeah it should be const &)

Yeah this problem is called `Longest common substring problem` so i named it as such, maybe something like `relativeSimilarity` would be better